app/controllers/application_controller.rb

```diff
@@ -1,6 +1,17 @@
 class ApplicationController < ActionController::Base
   protect_from_forgery with: :exception
   before_action :set_locale
+  after_filter :flash_to_headers
+
+  def flash_to_headers
+    return unless request.xhr?
+    response.headers['X-Message'] = flash[:error]  unless flash[:error].blank?
+    response.headers['X-Message'] = flash[:warning] unless flash[:warning].blank?
+    response.headers['X-Message'] = flash[:notice] unless flash[:notice].blank?
+    response.headers['X-Message'] = flash[:message] unless flash[:message].blank?
```

Comment:
Line is too long. [81/80]<br>Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.